code/math/vecmat.cpp

```diff
@@ -2954,6 +2954,54 @@ void vm_interpolate_angles_quick(angles *dest0, angles *src0, angles *src1, floa
 	dest0->b = src0->b + (arc_measures.b * interp_perc);
 }
 
+// Interpolate between two matrices, using t as a percentage progress between them.
+// derived by Asteroth from our AI code
+void vm_interpolate_matrices(matrix* out_orient, matrix* curr_orient, matrix* goal_orient, float t) 
+{
+	// check the two cases where it doesn't make sense to go through the whole function
+	if (fl_near_zero(t)) {
+		*out_orient = *curr_orient;
+		return;
+	}
+
+	Assertion(t <= 1.0f, "The function vm_interpolate_matrices was passed a bad percentage of %f.  This is an error that points to a coder mistake.  Please report to the SCP!",t);
+
+	// any value > 1.0f should be a bug.
+	if (t >= 1.0f) {
```

Comment:
I would handle this in the caller's context, t out of 0..1 is odd, but perfectly mathematically sensical, and there might be contexts where someone might want it.

Reply:
Okay.  Lucky I already do check it outside the funciton.